### PR TITLE
Move run_podfile_post_install_hooks call to execute right before projects are saved

### PR DESCRIPTION
**Pull request by yusufoos**
This will allow us to make changes to the project and schema without it getting overwritten by the pods_project_writer. #9379

### CODE REVIEW

**Comment by dnkoutso** [comment]
PR fails on `in runs the post-install hooks before serializing the Pods project`

Error:
```
Bacon::Error: not all expectations were satisfied
unsatisfied expectations:
- expected exactly once, not yet invoked: #<Pod::Installer:0x7c9b358>.run_podfile_post_install_hooks(any_parameters); in sequence 'hooks'
- expected exactly once, not yet invoked: #<AnyInstance:Pod::Installer::Xcode::PodsProjectWriter>.write!(any_parameters); in sequence 'hooks'
satisfied expectations:
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Move `run_podfile_post_install_hooks` call to execute right before projects are saved.
```
Comment (dnkoutso) [inline CHANGELOG.md]: please add two empty spaces at the end of this line for markdown formatting.

**Comment by yusufoos** [comment]
I have a dependent PR that fixes the failing integration test:
https://github.com/CocoaPods/cocoapods-integration-specs/pull/258

**Comment by yusufoos** [comment]
@dnkoutso Should be good to go now, all green.

**Comment by dnkoutso** [comment]
@yusufoos why did the integration specs had to change? [This](https://github.com/CocoaPods/cocoapods-integration-specs/compare/f53f3b354e586cbeb473c05995c98a20d29ec9e2...02b224fabc934a7fa6d119ab2572946adbe283b7#diff-c43121a98702cf7b7665215d89eb1322) feels like a breaking change to me?

**Comment by yusufoos** [comment]
The method `cleanup_projects` sorts the projects and is now called _before_ we run the the post install hooks. The integration test was assuming the "first" target to be "Reachability" which was the case when the targets were not sorted. Now that the targets are sorted the "first" target is "Pods-SampleApp". I changed the integration test so that it does not make any assumptions about the order of `installer.pods_project.targets`

**Comment by dnkoutso** [comment]
I see but can we not update all targets in that test and instead maintain it?
```ruby
  target = installer.pods_project.targets.find { |t| t.name == 'Reachability' }
  target.build_configurations.each do |config|
    config.build_settings['GCC_ENABLE_OBJC_GC'] = 'supported'
  end
```

**Comment by dnkoutso** [comment]
I am still skeptical about:
> The integration test was assuming the "first" target to be Reachability which was the case when the targets were not sorted. Now that the targets are sorted, the "first" target is Pods-SampleApp

An update to 1.9.x will break users who have made this assumption...

**Comment by yusufoos** [comment]
@dnkoutso I've updated the PR to move the sort call after post install hooks is called to ensure we don't break users who made assumptions about the ordering.

**Comment by dnkoutso** [comment]
Nice thanks I think that should be good. Sorry for being pedantic but we never know about how users use those :)

**Pull request merged**